In [2]:
import os
import pandas as pd

In [3]:
os.chdir("C:\\Users\\Sevak\\Desktop\\Analysis-of-Real-Estate-Markets-of-Armenia-main\\data")
os.listdir()

['cleaned_data.csv', 'encoded_data.csv', 'raw_data.csv']

In [4]:
data = pd.read_csv('raw_data.csv')
data.head(10)

Տարածաշրջան         Շինության տիպ Սենյակներ  Մակերես  Հարկ  \
0          Երևան, Շենգավիթ                  Քարե         3       50     1   
1           Երևան, Արաբկիր  Մոնոլիտ, Նորակառույց         2       70     3   
2  Երևան, Մալաթիա Սեբաստիա             Պանելային         3      100     7   
3  Երևան, Մալաթիա Սեբաստիա             Պանելային         1       45     1   
4  Երևան, Մալաթիա Սեբաստիա             Պանելային         3       85     6   
5   Երևան, Քանաքեռ Զեյթուն             Կասետային         2       52     4   
6  Երևան, Մալաթիա Սեբաստիա             Պանելային         1       45     2   
7           Երևան, Կենտրոն             Պանելային         3       80     9   
8           Երևան, Կենտրոն                  Քարե         1       31     4   
9           Երևան, Կենտրոն                  Քարե         3       77     4   

     Վերանորոգում     Գինը  
0         Կապիտալ   $64000  
1  Զրոյական վիճակ  $150000  
2   Պետական վիճակ  $115000  
3         Կապիտալ   $67000  
4   Պետական վիճակ     $386  
5    Նորմալ վիճակ   $68000  
6    Նորմալ վիճակ   $62000  
7    Նորմալ վիճակ  $168000  
8   Պետական վիճակ   $80000  
9    Նորմալ վիճակ  $163000

In [6]:
data = data.rename(
    {
        "Տարածաշրջան" : "Region",
        "Շինության տիպ" : "Building Type",
        "Սենյակներ" : "Rooms",
        "Մակերես" : "Area",
        "Հարկ" : "Floor",
        "Վերանորոգում" : "Renovation",
        "Գինը" : "Price"
    },
    axis = 1,
    inplace = False
)

In [7]:
data.shape

(4590, 7)

In [8]:
data.isnull().sum()

Region           0
Building Type    0
Rooms            0
Area             0
Floor            0
Renovation       0
Price            0
dtype: int64

In [9]:
data["Region"].value_counts()

Region
Երևան, Արաբկիր                                1232
Երևան, Կենտրոն                                 781
Երևան, Նոր Նորք                                494
Երևան, Քանաքեռ Զեյթուն                         392
Երևան, Մալաթիա Սեբաստիա                        347
Երևան, Դավթաշեն                                322
Երևան, Աջափնյակ                                314
Երևան, Շենգավիթ                                285
Երևան, Էրեբունի                                154
Երևան, Ավան                                    148
Կոտայք, Աբովյան                                 12
Երևան, Նորք Մարաշ                               11
Կոտայք, Հրազդան                                  9
Երևան, Նուբարաշեն                                9
Արարատ, Արարատ                                   8
Լոռի, Վանաձոր                                    7
Արարատ, Մասիս                                    6
Արմավիր, Էջմիածին քաղաք                          6
Արարատ, Արտաշատ                                  5
Արմավիր, Մեծամոր        

In [10]:
data = data[data["Region"].str.startswith("Երևան")]

In [11]:
data["Region"].value_counts()

Region
Երևան, Արաբկիր             1232
Երևան, Կենտրոն              781
Երևան, Նոր Նորք             494
Երևան, Քանաքեռ Զեյթուն      392
Երևան, Մալաթիա Սեբաստիա     347
Երևան, Դավթաշեն             322
Երևան, Աջափնյակ             314
Երևան, Շենգավիթ             285
Երևան, Էրեբունի             154
Երևան, Ավան                 148
Երևան, Նորք Մարաշ            11
Երևան, Նուբարաշեն             9
Name: count, dtype: int64

In [12]:
data[['City', 'District']] = data['Region'].str.split(', ', expand=True)

In [13]:
del data["Region"]
del data["City"]

In [14]:
data

Building Type Rooms  Area  Floor      Renovation    Price  \
0                     Քարե     3    50      1         Կապիտալ   $64000   
1     Մոնոլիտ, Նորակառույց     2    70      3  Զրոյական վիճակ  $150000   
2                Պանելային     3   100      7   Պետական վիճակ  $115000   
3                Պանելային     1    45      1         Կապիտալ   $67000   
4                Պանելային     3    85      6   Պետական վիճակ     $386   
...                    ...   ...   ...    ...             ...      ...   
4585                  Քարե     3    42      5         Կապիտալ   $70000   
4586             Պանելային     2    53      4    Նորմալ վիճակ   $82000   
4587  Մոնոլիտ, Նորակառույց     2    46      3    Դիզայներական  $148000   
4588             Պանելային     4    84      7   Պետական վիճակ  $142000   
4589                  Քարե     2    52      2        Կոսմետիկ   $83000   

              District  
0             Շենգավիթ  
1              Արաբկիր  
2     Մալաթիա Սեբաստիա  
3     Մալաթիա Սեբաստիա  
4     Մալաթիա Սեբաստիա  
...                ...  
4585          Էրեբունի  
4586          Աջափնյակ  
4587           Կենտրոն  
4588           Կենտրոն  
4589          Էրեբունի  

[4489 rows x 7 columns]

In [15]:
cols = ['District'] + [col for col in data.columns if col != 'District']
data = data[cols]

In [16]:
data.head()

District         Building Type Rooms  Area  Floor      Renovation  \
0          Շենգավիթ                  Քարե     3    50      1         Կապիտալ   
1           Արաբկիր  Մոնոլիտ, Նորակառույց     2    70      3  Զրոյական վիճակ   
2  Մալաթիա Սեբաստիա             Պանելային     3   100      7   Պետական վիճակ   
3  Մալաթիա Սեբաստիա             Պանելային     1    45      1         Կապիտալ   
4  Մալաթիա Սեբաստիա             Պանելային     3    85      6   Պետական վիճակ   

     Price  
0   $64000  
1  $150000  
2  $115000  
3   $67000  
4     $386

In [17]:
data["Rooms"].value_counts()

Rooms
3                     1910
2                     1578
1                      476
4                      448
5                       41
4, Դուպլեքս              8
6                        6
3, Դուպլեքս              6
2, Դուպլեքս              4
5, Դուպլեքս              4
Ազատ նախագիծ             3
7                        3
8 և ավել, Դուպլեքս       1
6, Դուպլեքս              1
Name: count, dtype: int64

In [18]:
values_to_delete = [
    "4, Դուպլեքս",
    "6",
    "3, Դուպլեքս",
    "2, Դուպլեքս",
    "5, Դուպլեքս",
    "Ազատ նախագիծ",
    "7",
    "8 և ավել, Դուպլեքս",
    "6, Դուպլեքս"
]
data = data[~data["Rooms"].isin(values_to_delete)]

In [19]:
data["Rooms"] = data["Rooms"].astype(int)

In [20]:
data.shape

(4453, 7)

In [21]:
Q1 = data["Area"].quantile(0.25)
Q3 = data["Area"].quantile(0.75)
IQR = Q3 - Q1

print(f"Q1 (25th percentile): {Q1}")
print(f"Q3 (75th percentile): {Q3}")
print(f"IQR (Interquartile Range): {IQR}")


Q1 (25th percentile): 53.0
Q3 (75th percentile): 84.0
IQR (Interquartile Range): 31.0


In [22]:
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

outliers = data[(data["Area"] < lower_bound) | (data["Area"] > upper_bound)]
print("Number of outliers:", outliers.shape[0])
outliers.shape

Number of outliers: 126


(126, 7)

In [23]:
data.shape

(4453, 7)

In [24]:
data = data[(data["Area"] >= lower_bound) & (data["Area"] <= upper_bound)]

In [25]:
data.shape

(4327, 7)

In [26]:
data['Price'] = (
    data['Price']
    .str.replace('$', '', regex=False)
    .str.replace(',', '', regex=False)
)
data['Price'] = pd.to_numeric(data['Price'], errors='coerce')

In [27]:
data = data.dropna(subset=['Price'])

In [28]:
data.isnull().sum()

District         0
Building Type    0
Rooms            0
Area             0
Floor            0
Renovation       0
Price            0
dtype: int64

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4276 entries, 0 to 4589
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   District       4276 non-null   object 
 1   Building Type  4276 non-null   object 
 2   Rooms          4276 non-null   int64  
 3   Area           4276 non-null   int64  
 4   Floor          4276 non-null   int64  
 5   Renovation     4276 non-null   object 
 6   Price          4276 non-null   float64
dtypes: float64(1), int64(3), object(3)
memory usage: 267.2+ KB


In [30]:
data.describe

<bound method NDFrame.describe of               District         Building Type  Rooms  Area  Floor  \
0             Շենգավիթ                  Քարե      3    50      1   
1              Արաբկիր  Մոնոլիտ, Նորակառույց      2    70      3   
2     Մալաթիա Սեբաստիա             Պանելային      3   100      7   
3     Մալաթիա Սեբաստիա             Պանելային      1    45      1   
4     Մալաթիա Սեբաստիա             Պանելային      3    85      6   
...                ...                   ...    ...   ...    ...   
4585          Էրեբունի                  Քարե      3    42      5   
4586          Աջափնյակ             Պանելային      2    53      4   
4587           Կենտրոն  Մոնոլիտ, Նորակառույց      2    46      3   
4588           Կենտրոն             Պանելային      4    84      7   
4589          Էրեբունի                  Քարե      2    52      2   

          Renovation     Price  
0            Կապիտալ   64000.0  
1     Զրոյական վիճակ  150000.0  
2      Պետական վիճակ  115000.0  
3            Կապի

In [31]:
data.shape

(4276, 7)

In [32]:
Q1_price = data["Price"].quantile(0.25)
Q3_price = data["Price"].quantile(0.75)
IQR_price = Q3_price - Q1_price

print(f"Q1 (25th percentile): {Q1_price}")
print(f"Q3 (75th percentile): {Q3_price}")
print(f"IQR (Interquartile Range): {IQR_price}")


Q1 (25th percentile): 82881.75
Q3 (75th percentile): 159000.0
IQR (Interquartile Range): 76118.25


In [33]:
lower_bound_price = Q1_price - 1.5 * IQR_price
upper_bound_price = Q3_price + 1.5 * IQR_price

print(f"Lower bound: {lower_bound_price}")
print(f"Upper bound: {upper_bound_price}")

Lower bound: -31295.625
Upper bound: 273177.375


In [34]:
data = data[(data["Price"] >= 30000) & (data["Price"] <= upper_bound_price)]
print(f"Shape after cleaning: {data.shape}")

Shape after cleaning: (4003, 7)


In [35]:
data.shape

(4003, 7)

In [36]:
data.to_csv(r"C:\\Users\\Sevak\\Desktop\\Analysis-of-Real-Estate-Markets-of-Armenia-main\\data\\cleaned_data.csv", index=False)

## Data Cleaning Summary

During this stage, I **cleaned and prepared the Yerevan real estate dataset** to ensure it is suitable for **accurate analysis and modeling**. Below is a clear summary of the cleaning steps:

---

### 1️ Initial Data Overview

* **Shape before cleaning:** `(4590, 7)` (4590 rows, 7 columns)
* Columns:

  * `District`, `Building Type`, `Rooms`, `Area`, `Floor`, `Renovation`, `Price`
* The dataset contained:

  * Inconsistent and rare category values
  * Potential outliers
  * Non-Yerevan entries not relevant for focused analysis.

---

### 2️ Column Cleaning

✅ **Renamed columns** from Armenian to English for clarity:

* `"Տարածաշրջան"` ➔ `"District"`, `"Սենյակներ"` ➔ `"Rooms"`, etc.

✅ **Filtered data to Yerevan only** to match the analysis scope.

✅ **Removed rare/inconsistent `Rooms` categories**, keeping only consistent, meaningful values:

* Removed values like `"Ազատ նախագիծ"`, `"4, Դուպլեքս"`, etc.
* Retained primary categories: `1`, `2`, `3`, `4`, `5`.

✅ Ensured **numeric columns** (`Area`, `Price`) were correctly converted to numeric types for analysis.

---

### 3️ Outlier Removal (`Area` and `Price`)

✅ Applied **IQR method on `Area`**:

* **Q1 (25%):** 53.0
* **Q3 (75%):** 84.0
* **IQR:** 31.0
* **Lower Bound:** 6.5
* **Upper Bound:** 130.5
* Removed **126 outlier rows** outside this range.

✅ Applied **upper-bound filtering on `Price`**:

* Used **IQR for upper bound** while manually setting a practical **lower bound (`~31,000`)** to avoid losing valuable data.
* Removed unrealistically high or low prices while preserving the market distribution.

---

### 4️ Final Result

✅ **Shape after cleaning:** `(4003, 7)`
✅ In total, **587 rows were removed** due to:

* Filtering non-Yerevan entries
* Removing rare/inconsistent category values
* Removing outliers in `Area` and `Price`.

---

###  Conclusion

The dataset is now:

* **Clean and consistent**
* **Focused on Yerevan**
* **Outlier-free for reliable modeling**

Ready for:

* Price prediction modeling
* Exploratory data analysis
* Generating clear visual insights into the **Yerevan real estate market**.

---
##  Key Cleaning Statistics

| Step                        | Description                                              | Rows Removed | Shape After Step |
| --------------------------- | -------------------------------------------------------- | ------------ | ---------------- |
| **Initial Data**            | Raw dataset                                              | -            | `(4590, 7)`      |
| **Filter to Yerevan**       | Removed non-Yerevan entries                              | 64           | `(4526, 7)`      |
| **Remove rare `Rooms`**     | Dropped inconsistent/rare `Rooms` categories             | 123          | `(4403, 7)`      |
| **Remove `Area` outliers**  | Using IQR method (`6.5 < Area < 130.5`)                  | 126          | `(4277, 7)`      |
| **Remove `Price` outliers** | Applied upper-bound IQR, set lower bound at \~31,000 AMD | 274          | `(4003, 7)`      |

 **Total rows removed:** `4590 - 4003 = 587`

 **Final dataset shape:** `(4003, 7)`

---

## Next Steps

The cleaned dataset is ready for:

* Exploratory Data Analysis (EDA)
* Feature engineering
* Predictive modeling for **Yerevan real estate prices**
